src/ssl.c

```diff
@@ -34814,14 +34817,33 @@ int wolfSSL_ECDH_compute_key(void *out, size_t outlen,
     }
 
     len = (word32)outlen;
+    key = (ecc_key*)ecdh->internal;
+
+#if defined(ECC_TIMING_RESISTANT) && !defined(HAVE_SELFTEST) \
+    && !defined(HAVE_FIPS)
+    if (key->rng == NULL) {
+        if (initGlobalRNG == 0 && wolfSSL_RAND_Init() != WOLFSSL_SUCCESS) {
+            WOLFSSL_MSG("No RNG to use");
+            return WOLFSSL_FATAL_ERROR;
+        }
+        key->rng = &globalRNG;
```

Comment:
Why the global?  Won't wolfSSL_EVP_PKEY_free() or some other free end up free'ing the globalRNG if something goes wrong before you "unset" it?

Reply:
I'll modify unsetting so that it always occurs after the wc_ecc_shared_secret_ssh call.